### PR TITLE
Refactor config

### PR DESCRIPTION
**Pull request by dougwig**

### CODE REVIEW

File: a10_neutron_lbaas/tests/unit/test_a10_config.py [hunk 1]
```diff
@@ -18,36 +18,36 @@
 class TestA10Config(test_base.UnitTestBase):
 
     def test_verify_appliances(self):
-        self.assertTrue(self.a.config.verify_appliances)
+        self.assertTrue(self.a.config.get('verify_appliances'))
 
     def test_num_appliances(self):
         # Everytime we update the test config, this test has to be updated
```
Comment (mdurrant-b3) [inline a10_neutron_lbaas/tests/unit/test_a10_config.py]: My useless commentary here could go, but it's just comments.

File: a10_neutron_lbaas/a10_config.py [hunk 2]
```diff
@@ -41,29 +41,29 @@ def __init__(self, config_dir=None):
             d = '/etc/neutron/services/loadbalancer/a10networks'
         else:
             d = '/etc/a10'
-        self.config_dir = os.environ.get('A10_CONFIG_DIR', d)
-        self.config_path = os.path.join(self.config_dir, "config.py")
+        self._config_dir = os.environ.get('A10_CONFIG_DIR', d)
```
Comment (mdurrant-b3) [inline a10_neutron_lbaas/a10_config.py]: According to Doug, order should be:
```passed in
A10_CONFIG_DIR
venv
crappy legacy path
/etc/a10```
Currently is:
```A10_CONFIG_DIR` env
`/etc/a10` dir
`/etc/a10/neutron/services/loadbalancer/a10networks` dir
`venv` dir
passed in `config_dir` dir```

**Comment by dougwig** [comment]
check

File: a10_neutron_lbaas/a10_config.py (hunk 2 above)
Comment (Cedev) [inline a10_neutron_lbaas/a10_config.py]: This code chooses the first of:
A10_CONFIG_DIR
passed in
venv
crappy legacy path
/etc/a10

**Comment by Cedev** [comment]
-1 Removing `.config` from A10Config will require everything that looks in config for something to have foll-blown a10-config support for it. This isn't bad for anything in a10-neutron-lbaas, but it's bad for extensions like a10-openstack that don't have a way to modify what a10-neutron-lbaas's A10Config class does.

**Comment by dougwig** [comment]
@Cedev, umm, yes, not seeing .config is the entire point.  a10-openstack should be instantiating an A10Config() object, which can access the get methods. It's to make the per provider config transparent to the higher-level code.

**Comment by dougwig** [comment]
Actually the order in this pull is:

A10_CONFIG_DIR
config_dir
venv
legacy
/etc/a10

**Comment by mdurrant-b3** [comment]
:+1:

**Pull request merged**